homeassistant/components/slide/__init__.py

```diff
@@ -0,0 +1,159 @@
+    """Set up the Slide platform."""
+
+    # pylint: disable=unused-argument
+    async def update_slides(now=None):
+        """Update slide information."""
+        result = await hass.data[DOMAIN][API].slidesoverview()
+
+        if result is None:
+            _LOGGER.error("Slide API does not work or returned an error")
+            return
+
+        if result:
+            _LOGGER.debug("Slide API returned %d slide(s)", len(result))
+        else:
+            _LOGGER.warning("Slide API returned 0 slides")
+
+        for key in hass.data[DOMAIN][SLIDES]:
+            hass.data[DOMAIN][SLIDES][key]["pos"] = None
+
+        for slide in result:
+            if "device_id" not in slide:
+                _LOGGER.error(
+                    "Found invalid Slide entry, 'device_id' is " "missing. Entry=%s",
+                    slide,
+                )
+                continue
+
+            uid = slide["device_id"].replace("slide_", "")
+            slidenew = {}
+            slidenew["mac"] = uid
+            slidenew["id"] = slide["id"]
+            slidenew["name"] = slide["device_name"]
+
+            if "device_info" not in slide:
+                _LOGGER.error(
+                    "Slide %s (%s) has no 'device_info' Entry=%s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide,
+                )
+                continue
+
+            # Check if we have 'pos' (OK) or 'code' (NOK)
+            if "pos" in slide["device_info"]:
+                slidenew["online"] = True
+                slidenew["pos"] = slide["device_info"]["pos"]
+                if slidenew["pos"] < 0:
+                    slidenew["pos"] = 0
+                elif slidenew["pos"] > 1:
+                    slidenew["pos"] = 1
+            elif "code" in slide["device_info"]:
+                slidenew["online"] = False
+                _LOGGER.warning(
+                    "Slide %s (%s) is offline with " "code=%s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide["device_info"]["code"],
+                )
+            else:
+                slidenew["online"] = False
+                _LOGGER.error(
+                    "Slide %s (%s) has invalid 'device_info'" " %s",
+                    slide["id"],
+                    slidenew["mac"],
+                    slide["device_info"],
+                )
+
+            hass.data[DOMAIN][SLIDES][uid] = slidenew
+            _LOGGER.debug("Updated entry=%s", slidenew)
+
+    if DOMAIN not in config:
+        return True
+
+    hass.data[DOMAIN] = {}
+    hass.data[DOMAIN][SLIDES] = {}
+
+    username = config[DOMAIN][CONF_USERNAME]
+    password = config[DOMAIN][CONF_PASSWORD]
+    scaninterval = config[DOMAIN][CONF_SCAN_INTERVAL]
+    timeout = config[DOMAIN][CONF_TIMEOUT]
+    retry = config[DOMAIN][CONF_RETRY]
+
+    from goslideapi import GoSlideCloud
+
+    hass.data[DOMAIN][API] = GoSlideCloud(username, password, timeout)
```

Comment:
Please have `GoSlideCloud` accept an optional aiohttp session. We want all integration that use aiohttp client to use the home assistant aiohttp session. Use the aiohttp client helper to get the session instance.

Reply:
At this moment the goslideapi doesn't use the aiohttp session, it uses the simple "aiohttp.request" version. I will have a look if i can change/improve this in the next version.

Reply:
I will resolve this, because I don't use an aiohttp.session at this moment.